architecture/understanding-development.adoc

```diff
@@ -1,45 +1,396 @@
+
+You have an idea for an application and you want to containerize it.
+
+First you require a tool for building a container, like buildah or docker,
+and a file that describes what goes in your container, which is typically, a
+link:https://docs.docker.com/engine/reference/builder/[Dockerfile].
+
+Next, you require a location to push the resulting container image so you can
+pull it to run anywhere you want it to run. This location is a container
+registry.
+
+Some examples of each of these components are installed by default on most
+Linux operating systems, except for the Dockerfile, which you provide yourself.
+
+The following diagram displays the process of building and pushing an image:
+
+.Create a simple containerized application and push it to a registry
+image::create-push-app.png[Creating and pushing a containerized application]
+
+If you use a computer that runs Red Hat Enterprise Linux (RHEL) as the operating
+system, the process of creating a containerized application requires the
+following steps:
+
+. Install container build tools: RHEL contains a set of tools that includes
+podman, buildah, and skopeo that you use to build and manage containers.
+. Create a Dockerfile to combine base image and software: Information about
+building your container goes into a file that is named `Dockerfile`. In that
+file, you identify the base image you build from, the software packages you
+install, and the software you copy into the container. You also identify
+parameter values like network ports that you expose outside the container and
+volumes that you mount inside the container. Put your Dockerfile and the
+software you want to containerized in a directory on your RHEL system.
+. Run buildah or docker build: Run the `buildah build-using-dockerfile` or
+the `docker build` command to pull you chosen base image to the local system and
+creates a container image that is stored locally. You can also build container
+without a Dockerfile by using buildah.
+. Tag and push to a registry: Add a tag to your new container image that
+identifies the location of the registry in which you want to store and share
+your container. Then push that image to the registry by running the
+`podman push` or `docker push` command.
+. Pull and run the image: From any system that has a container client tool,
+such as podman or docker, run a command that identifies your new image.
+For example, run the `podman run <image_name>` or `docker run <image_name>`
+command. Here `<image_name>` is the name of your new container image, which
+resembles `quay.io/myrepo/myapp:latest`. The registry might require credentials
+to push and pull images.
+
+For more details on the process of building container images, pushing them to
+registries, and running them, see
+link:https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux_atomic_host/7/html-single/managing_containers/index%23using_podman_to_work_with_containers[Using podman to work with containers]
```

Comment:
Both of these links are broken, but for "Building container images with buildah" we could replace the link with:

https://docs.openshift.com/container-platform/4.1/builds/custom-builds-buildah.html